### DISCUSSION TITLE
onAfterBootstrap

### DISCUSSION THREAD

**Discussion by einarpersson** [pocketbase/pocketbase#3220] [Q&A]
I am looking at the docs

> onAfterBootstrap hook is triggered after initializing the main application resources (eg. after db open and initial settings load).

1. Will migrations run **before** or **after** `onAfterBootstrap`?
2. If migrations run **after**, is there another hook that marks that all migrations have run and _**everything**_ related to the initialization process has completed?

**EDIT**
I notice that there is a `onBeforeServe` in Go-hooks but it is missing in JS-hooks. Is it possible to add it to JS-hooks as well?

**Comment by ganigeorgiev** [comment]
1. The migrations runs after since they need the initialized db from the bootstrap process.

2. I guess the [`OnBeforeServe`](https://pocketbase.io/docs/go-event-hooks/#onbeforeserve) hook could be used for that but as you noted it is not available in the JS `pb_hooks` context, at least not directly.
  If you use the JS hooks, one slightly hacky workaround could be to attach to the hook through the `$app` instance, but note that you can't use the hook to register custom routes as in Go. For this you have to use the dedicated `routerAdd()`, `routerUse()` etc. handlers (_the reason for this limitation could be found in https://github.com/pocketbase/pocketbase/discussions/2825#discussioncomment-6376210_).
    ```js
    $app.onBeforeServe().add(() => {
      console.log('before serve')
    });
    ```
